extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +38,129 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
     /** @private {?Element} */
     this.element_ = element;
 
-    /** @private {?Element} */
-    this.image_ = null;
+    /** @private {?Element} The element that is transitioned. */
+    this.ampImgEl_ = null;
 
-    /** @private {?string} */
+    /** @private {?string} Sent to siblings to update their position. */
     this.x_ = null;
 
-    /** @private {?string} */
+    /** @private {?string} Sent to siblings to update their position. */
     this.y_ = null;
 
-    /** @private {?string} */
+    /** @private {?string} Sent to siblings to update their position. */
     this.zoom_ = null;
+
+    /** @private {?../../../extensions/amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {boolean} */
+    this.isOnActivePage_ = false;
   }
 
   /** @override */
   buildCallback() {
+    this.ampImgEl_ = dev().assertElement(
+      this.element_.querySelector('amp-img')
+    );
+
     this.x_ = this.element_.getAttribute('x') || '0%';
     this.y_ = this.element_.getAttribute('y') || '0%';
     this.zoom_ = this.element_.getAttribute('zoom') || '1';
+
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        this.storeService_ = storeService;
+      }
+    );
   }
 
   /** @override */
   layoutCallback() {
-    const ampImgEl = dev().assertElement(
-      this.element_.querySelector('amp-img')
-    );
-    return whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    this.initializeListeners_();
+    return whenUpgradedToCustomElement(this.ampImgEl_)
+      .then(() => this.ampImgEl_.signals().whenSignal(CommonSignals.LOAD_END))
       .then(() => {
-        this.image_ = dev().assertElement(this.element.querySelector('img'));
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setStyles(this.ampImgEl_, {
```

Comment:
Do these need to be overridable by publishers? If not, we should use `setImportantStyles`